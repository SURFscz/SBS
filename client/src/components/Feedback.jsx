import React, {useEffect, useRef, useState} from "react";
import Modal from "react-modal";
import I18n from "i18n-js";
import {ReactComponent as InformationIcon} from "../icons/informational.svg";
import "./Feedback.scss";
import Button from "./Button";
import {isEmpty} from "../utils/Utils";
import {feedback} from "../api";
import {setFlash} from "../utils/Flash";
import DOMPurify from "dompurify";

export default function FeedbackDialog({isOpen = false, close}) {

    const [message, setMessage] = useState("");

    const inputRef = useRef(null);

    useEffect(() => {
        inputRef.current && inputRef.current.focus();
    });

    const doClose = () => {
        close();
        setTimeout(() => setMessage(""), 500);
    }

    const sendFeedBack = () => {
        feedback(message).then(() => {
            doClose();
            setFlash(I18n.t("feedback.flash"));
        });
    }

    return (
        <Modal
            isOpen={isOpen}
            onRequestClose={doClose}
            className="feedback-dialog-content"
            overlayClassName="feedback-dialog-overlay"
            closeTimeoutMS={250}
            ariaHideApp={false}>
            <h1>{I18n.t("feedback.title")}</h1>
            <section className="info">
                <InformationIcon/>
                <span dangerouslySetInnerHTML={{__html: DOMPurify.sanitize(I18n.t("feedback.info"))}}/>
            </section>
            <section className="feedback">
                <textarea name="feedback"
                          id="feedback"
                          value={message}
                          rows="10"
                          ref={inputRef}
                          onChange={e => setMessage(e.target.value)}/>
            </section>
            <section className="help">
                <h2 className="title" dangerouslySetInnerHTML={{__html: DOMPurify.sanitize(I18n.t("feedback.help"))}}/>
                <span dangerouslySetInnerHTML={{__html: DOMPurify.sanitize(I18n.t("feedback.helpInfo"))}}/>
            </section>
            <section className="disclaimer">
                <span dangerouslySetInnerHTML={{__html: DOMPurify.sanitize(I18n.t("feedback.disclaimer"))}}/>
            </section>
            <section className="actions">
                <Button txt={I18n.t("forms.cancel")} cancelButton={true} onClick={doClose}/>
                <Button txt={I18n.t("feedback.send")} disabled={isEmpty(message)} onClick={sendFeedBack}/>
            </section>

        </Modal>
    );

}

