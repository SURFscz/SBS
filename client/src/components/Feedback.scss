@import "../stylesheets/vars.scss";


.feedback-dialog-overlay {
    position: fixed;
    display: flex;
    align-items: center;
    justify-content: center;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background-color: rgba(255, 255, 255, 0.85);
    z-index: 5;

    .feedback-dialog-content {
        padding: 40px 60px;

        @media (max-width: $compact-medium) {
            padding: 20px;
        }


        background-color: white;
        overflow: auto;
        border-radius: $br;
        outline: none;
        z-index: 5;
        display: flex;
        flex-direction: column;
        border: 1px solid $lighter-grey;
        box-shadow: 1px 2px 2px 0 rgba(0, 0, 0, 0.3);

        h1 {
            font-size: 32px;
            line-height: 36px;
            margin-bottom: 15px;
        }

        h2 {
            font-size: 20px;
            line-height: 24px;
            margin: 15px 0 10px 0;

            &.title {
                margin-bottom: 3px;
            }
        }

        section.info {
            display: flex;
            align-items: center;
            border-radius: 4px;
            padding: 15px;
            margin-bottom: 15px;

            svg {
                margin-right: 15px;
            }

            background-color: #95d7e4;
        }

        textarea {
            display: block;
            width: 100%;
            border: 1px solid $dark-grey;
            padding: 6px 12px;
            border-radius: $br;
            font-size: 16px;
            margin-bottom: 10px;
            &:focus {
                outline: none;
                box-shadow: 3px 3px 3px $blue-hover, -3px -3px 1px $blue-hover;
            }

        }

        section.disclaimer {
            font-size: 13px;
            margin-top: 25px;
        }

    }
}

.ReactModalPortal .feedback-dialog-overlay {
    transition: opacity 250ms ease-in-out;

    &--before-close {
        opacity: 0;
    }
}

.ReactModalPortal .feedback-dialog-content {
    transition: top 250ms ease-in-out, bottom 250ms ease-in-out;
    width: 660px;
    @media (max-width: $compact-medium) {
        width: 280px;
    }

    &--before-close {
        top: -50%;
        bottom: 100%;
    }
}
