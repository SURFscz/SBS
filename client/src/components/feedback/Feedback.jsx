import React, {useEffect, useRef, useState} from "react";

import I18n from "../../locale/I18n";
import "./Feedback.scss";
import {isEmpty} from "../../utils/Utils";
import {feedback} from "../../api";
import {setFlash} from "../../utils/Flash";
import DOMPurify from "dompurify";
import {AlertType, Modal} from "@surfnet/sds";

export default function FeedbackDialog({isOpen = false, close, initialMessage = ""}) {

    if (!isOpen) {
        return null;
    }

    const [message, setMessage] = useState(initialMessage);

    const inputRef = useRef(null);

    useEffect(() => {
        inputRef.current && inputRef.current.focus();
    });

    const doClose = () => {
        close();
        setTimeout(() => setMessage(""), 500);
    }

    const sendFeedBack = () => {
        feedback(message).then(() => {
            doClose();
            setFlash(I18n.t("feedback.flash"));
        });
    }

    const content = () => {
        return (
            <div className={"feedback-content"}>
                <div className="sds--text-area">
                    <textarea name="feedback"
                              id="feedback"
                              value={message}
                              rows="10"
                              ref={inputRef}
                              onChange={e => setMessage(e.target.value)}/>
                </div>
                <section className="help">
                    <h3 className="title"
                        dangerouslySetInnerHTML={{__html: DOMPurify.sanitize(I18n.t("feedback.help"))}}/>
                    <span dangerouslySetInnerHTML={{__html: DOMPurify.sanitize(I18n.t("feedback.helpInfo"))}}/>
                </section>
                <section className="disclaimer">
                <span dangerouslySetInnerHTML={{
                    __html: DOMPurify.sanitize(I18n.t("feedback.disclaimer"),
                        {ADD_ATTR: ['target']})
                }}/>
                </section>
            </div>)
    }


    return (
        <Modal
            confirm={sendFeedBack}
            cancel={doClose}
            alertType={AlertType.Info}
            subTitle={I18n.t("feedback.info")}
            children={content()}
            title={I18n.t("feedback.title")}
            cancelButtonLabel={I18n.t("forms.cancel")}
            confirmationButtonLabel={I18n.t("feedback.send")}
            confirmDisabled={isEmpty(message)}
            question={null}/>
    );

}
