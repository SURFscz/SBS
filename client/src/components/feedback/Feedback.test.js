import React from "react";
import { render } from "@testing-library/react";
import Feedback from "./Feedback";

describe("Feedback", () => {
  it("matches snapshot with default props", () => {
    const { asFragment } = render(<Feedback />);
    expect(asFragment()).toMatchSnapshot();
  });
});
