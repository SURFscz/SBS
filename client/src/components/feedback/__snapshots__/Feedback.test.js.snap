// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback matches snapshot with default props 1`] = `<DocumentFragment />`;
